cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -853,3 +840,129 @@ func (ht *hcsTask) Stats(ctx context.Context) (*stats.Statistics, error) {
 	}
 	return s, nil
 }
+
+func (ht *hcsTask) Update(ctx context.Context, req *task.UpdateTaskRequest) error {
+	resources, err := typeurl.UnmarshalAny(req.Resources)
+	if err != nil {
+		return errors.Wrapf(err, "failed to unmarshal resources for container %s update request", req.ID)
+	}
+
+	if err := verifyTaskUpdateResourcesType(resources); err != nil {
+		return err
+	}
+
+	if ht.ownsHost && ht.host != nil {
+		return ht.host.UpdateConstraints(ctx, resources, req.Annotations)
+	}
+
+	return ht.updateTaskContainerResources(ctx, resources, req.Annotations)
+}
+
+func (ht *hcsTask) updateTaskContainerResources(ctx context.Context, data interface{}, annotations map[string]string) error {
```

Comment:
Looks like the the `LinuxResources` struct has a few other fields than `CPU` & `Memory` (similarly the `WindowsResources` field has one other field). Do we not care about these fields? Especially for LCOW because we forward the update request as it is in case of LCOW. What if that request contains some other fields than CPU & Memory?

Reply:
LCOW already supports just passing everything through to runc to be updated and should work as expected. For WCOW though we need to have the support in HCS to actually update the resource. There are some other fields that are supported by hcs, but this initial work was for the high pri fields.